plugin/src/main/cpp/extensions/openxr_fb_space_warp_extension_wrapper.cpp

```diff
@@ -0,0 +1,252 @@
+	for (int i = 0; i < view_count; i++) {
+		space_warp_info[i].type = XR_TYPE_COMPOSITION_LAYER_SPACE_WARP_INFO_FB;
+
+		space_warp_info[i].next = nullptr;
+
+		space_warp_info[i].layerFlags = 0;
+
+		space_warp_info[i].motionVectorSubImage.swapchain = (XrSwapchain)get_openxr_api()->openxr_swapchain_get_swapchain(motion_vector_swapchain_info);
+		space_warp_info[i].motionVectorSubImage.imageRect.offset.x = 0;
+		space_warp_info[i].motionVectorSubImage.imageRect.offset.y = 0;
+		space_warp_info[i].motionVectorSubImage.imageRect.extent.width = system_space_warp_properties.recommendedMotionVectorImageRectWidth;
+		space_warp_info[i].motionVectorSubImage.imageRect.extent.height = system_space_warp_properties.recommendedMotionVectorImageRectHeight;
+		space_warp_info[i].motionVectorSubImage.imageArrayIndex = i;
+
+		space_warp_info[i].appSpaceDeltaPose = { { 0.0, 0.0, 0.0, 1.0 }, { 0.0, 0.0, 0.0 } };
+
+		space_warp_info[i].depthSubImage.swapchain = (XrSwapchain)get_openxr_api()->openxr_swapchain_get_swapchain(motion_vector_depth_swapchain_info);
+		space_warp_info[i].depthSubImage.imageRect.offset.x = 0;
+		space_warp_info[i].depthSubImage.imageRect.offset.y = 0;
+		space_warp_info[i].depthSubImage.imageRect.extent.width = system_space_warp_properties.recommendedMotionVectorImageRectWidth;
+		space_warp_info[i].depthSubImage.imageRect.extent.height = system_space_warp_properties.recommendedMotionVectorImageRectHeight;
+		space_warp_info[i].depthSubImage.imageArrayIndex = i;
+
+		space_warp_info[i].minDepth = 0.0;
+		space_warp_info[i].maxDepth = 1.0;
+
+		space_warp_info[i].farZ = get_openxr_api()->get_render_state_z_near();
+		space_warp_info[i].nearZ = get_openxr_api()->get_render_state_z_far();
+	}
+}
+
+void OpenXRFbSpaceWarpExtensionWrapper::_on_pre_render() {
+	if (!fb_space_warp_ext || !enabled) {
+		return;
+	}
+
+	get_openxr_api()->openxr_swapchain_acquire(motion_vector_swapchain_info);
+	get_openxr_api()->openxr_swapchain_acquire(motion_vector_depth_swapchain_info);
+
+	RID motion_vector_swapchain_image = get_openxr_api()->openxr_swapchain_get_image(motion_vector_swapchain_info);
+	get_openxr_api()->set_velocity_texture(motion_vector_swapchain_image);
+	RID motion_vector_depth_swapchain_image = get_openxr_api()->openxr_swapchain_get_image(motion_vector_depth_swapchain_info);
+	get_openxr_api()->set_velocity_depth_texture(motion_vector_depth_swapchain_image);
+
+	int target_width = system_space_warp_properties.recommendedMotionVectorImageRectWidth;
+	int target_height = system_space_warp_properties.recommendedMotionVectorImageRectHeight;
+	Size2i render_target_size = { target_width, target_height };
+	get_openxr_api()->set_velocity_target_size(render_target_size);
+
+	Transform3D world_transform = XRServer::get_singleton()->get_world_origin();
+	Transform3D delta_transform = render_state.previous_transform.affine_inverse() * world_transform;
+	Quaternion delta_quat = delta_transform.basis.get_quaternion();
```

Comment:
I hate smooth rotation in VR, but we should probably have it in the sample, because otherwise we're not testing that `delta_quat` is accurate

Reply:
Smooth rotation mode added!